# HG changeset patch
# User Konstantinos Poulios  <logari81@googlemail.com>
# Date 1331851792 14400
#      Thu Mar 15 18:49:52 2012 -0400
# Node ID d16638f73691312020556f6cb5f3b004f10ca4fc
# Parent  ec7ab7e37731edd3f825ed3f64dc0e6957153304
Modify gl2ps_renderer.cc to properly specificy 2D/3D text object positions.
Regression of changeset 7a49519913e3.  Bug # 34907.

* gl2ps_renderer.cc (gl2ps_renderer::draw_text):

diff --git a/src/gl2ps-renderer.cc b/src/gl2ps-renderer.cc
--- a/src/gl2ps-renderer.cc
+++ b/src/gl2ps-renderer.cc
@@ -212,17 +212,17 @@ glps_renderer::draw_text (const text::pr
     valign = 2;
   else if (props.verticalalignment_is ("baseline"))
     valign = 3;
   else if (props.verticalalignment_is ("middle"))
     valign = 1;
 
   // FIXME: handle margin and surrounding box
 
-  glRasterPos3d (pos(0), pos(1), pos(2));
+  glRasterPos3d (pos(0), pos(1), pos.numel () > 2 ? pos(2) : 0.0);
 
   octave_value string_prop = props.get_string ();
 
   string_vector sv = string_prop.all_strings ();
 
   std::string s = sv.join ("\n");
 
   gl2psTextOpt (s.c_str (), fontname.c_str (), fontsize,
